src/Core/src/Platform/Android/SafeAreaExtensions.cs

```diff
@@ -248,27 +234,28 @@ internal static double GetSafeAreaForEdge(SafeAreaRegions safeAreaRegion, double
     {
         // Edge-to-edge content - no safe area padding
         if (safeAreaRegion == SafeAreaRegions.None)
-        {
             return 0;
-        }
 
-        // Handle SoftInput specifically - only apply keyboard insets for bottom edge when keyboard is showing
-        if (isKeyboardShowing && edge == 3)
+        // Handle SoftInput specifically for bottom edge when keyboard is showing
+        if (edge == 3 && SafeAreaEdges.IsSoftInput(safeAreaRegion))
         {
-            if (SafeAreaEdges.IsSoftInput(safeAreaRegion))
-                return keyBoardInsets.Bottom;
-
-            // if they keyboard is showing then we will just return 0 for the bottom inset
-            // because that part of the view is covered by the keyboard so we don't want to pad the view
-            return 0;
+            return HandleSoftInputRegion(safeAreaRegion, originalSafeArea, isKeyboardShowing, keyBoardInsets);
         }
 
-        // All other regions respect safe area in some form
-        // This includes:
-        // - Default: Platform default behavior
-        // - All: Obey all safe area insets  
-        // - Container: Content flows under keyboard but stays out of bars/notch
-        // - Any combination of the above flags
+        // All other regions respect safe area (Default, All, Container, etc.)
         return originalSafeArea;
     }
+
+    static double HandleSoftInputRegion(SafeAreaRegions safeAreaRegion, double originalSafeArea, bool isKeyboardShowing, SafeAreaPadding keyBoardInsets)
+    {
+        if (!isKeyboardShowing)
+        {
+            // When keyboard is hidden, only apply safe area for "All" regions
+            return safeAreaRegion == SafeAreaRegions.All ? originalSafeArea : 0;
+        }
+
+        // When keyboard is showing, use the larger of keyboard inset or original safe area
+        var keyboardInset = keyBoardInsets.Bottom;
+        return Math.Max(keyboardInset, originalSafeArea);
```

Comment:
Previously just returned `keyboardInset`. With floating/split keyboards, the keyboard inset might be smaller than the navigation bar safe area. Could verify?